src/state/selectors.ts

```diff
@@ -88,3 +98,6 @@ export const selectTxStatusParamFromFilter = selectParamFromFilter('txStatus')
 export const selectFeeModelParamFromFilter = selectParamFromFilter('feeModel')
 export const selectTimeframeParamFromFilter = selectParamFromFilter('timeframe')
 export const selectOnlyConnectedChainsParamFromFilter = selectParamFromFilter('onlyConnectedChains')
+
+export const selectHopIndexParamFromRequiredFilter = selectRequiredParamFromFilter('hopIndex')
+export const selectTradeIdParamFromRequiredFilter = selectRequiredParamFromFilter('tradeId')
```

Comment:
q: do the current typings of `selectXyzParamFromFilter` make the param optional?

Reply:
They do indeed, which lead to many downstream problems is the `selectRequiredParamFromFilter` version is not used. This one will ensure they are passed to the selector at a type level.

Reply:
Beautiful. Could definitely leverage that for other filters as well, I was always under the impression this was strongly typed and made those params required!